src/Features/Core/Portable/PullMemberUp/QuickAction/AbstractMemberPullerWithQuickAction.cs

```diff
@@ -0,0 +1,187 @@
+                        // When it is a property, it could have a public getter/setter
+                        // but other one is not public. In this scenario, only the public getter/setter
+                        // will be add to the destination interface
+                        return CodeGenerationSymbolFactory.CreatePropertySymbol(
+                            propertySymbol,
+                            getMethod: FilterGetterOrSetter(propertySymbol.GetMethod),
+                            setMethod: FilterGetterOrSetter(propertySymbol.SetMethod));
+                    }
+                    else
+                    {
+                        return analysisResult.Member;
+                    }
+                });
+            var codeGenerationService = contextDocument.Project.LanguageServices.GetRequiredService<ICodeGenerationService>();
+            var options = new CodeGenerationOptions(generateMethodBodies: false, generateMembers: false);
+            return await codeGenerationService.AddMembersAsync(
+                contextDocument.Project.Solution, result.Destination, symbolsToPullUp, options: options, cancellationToken: cancellationToken);
+        }
+
+        private Solution PullMembersIntoClass(
+            PullMembersUpAnalysisResult result,
+            ICodeGenerationService codeGenerationService,
+            SyntaxNode destinationNodeSyntax,
+            SolutionEditor solutionEditor,
+            Solution solution,
+            ImmutableDictionary<SyntaxTree, DocumentEditor> editorMap,
+            ImmutableDictionary<ISymbol, ImmutableArray<SyntaxNode>> syntaxMap)
+        {
+            // Add members to destination
+            var pullUpMembersSymbols = result.MembersAnalysisResults.SelectAsArray(memberResult => memberResult.Member);
+            var options = new CodeGenerationOptions(reuseSyntax: true, generateMethodBodies: false);
+            var membersAddedNode = codeGenerationService.AddMembers(destinationNodeSyntax, pullUpMembersSymbols, options: options);
+            var destinationEditor = editorMap[destinationNodeSyntax.SyntaxTree];
+            destinationEditor.ReplaceNode(destinationNodeSyntax, (syntaxNode, generator) => membersAddedNode);
+
+            // Remove the original members since we are pulling members into class
+            foreach (var analysisResult in result.MembersAnalysisResults)
+            {
+                foreach (var syntax in syntaxMap[analysisResult.Member])
+                {
+                    var originalMemberEditor = editorMap[syntax.SyntaxTree];
+                    originalMemberEditor.RemoveNode(originalMemberEditor.Generator.GetDeclaration(syntax));
+                }
+            }
+
+            return solutionEditor.GetChangedSolution();
+        }
+
+        private async Task<(ImmutableDictionary<SyntaxTree, DocumentEditor> editorMap, ImmutableDictionary<ISymbol, ImmutableArray<SyntaxNode>> syntaxMap)> InitializeEditorMapsAndSyntaxMapAsync(
```

Comment:
consider making some sort of alias for this type, or just a new type.  When you're tuple itself is so long that just the *tuple* can't be seen on an entire line of text... that's difficult to read :)

Reply:
note: since you can't alias a tuple, you could make aliases for the two dictionary types you're using inside the tuple.  i.e. `using EditorMap = ...`